src/openfermion/ops/_code_operator.py

```diff
@@ -0,0 +1,298 @@
+import copy
+
+import numpy
+import scipy
+import scipy.sparse
+
+from openfermion.ops import SymbolicBinary
+
+
+def _shift_decoder(decode, shift_constant):
+    """ Shifts the indices of a decoder by a constant.
+    
+    Args:
+        decode (list of SymbolicBinary): a decoder
+        shift_constant (int): the qubit index that corresponds to the offset.
+
+    Returns (list of SymbolicBinary):  shifted decoder
+    """
+    decode_shifted = []
+    for entry in decode:
+        tmp_entry = copy.deepcopy(entry)
+        tmp_entry._shift(shift_constant)
+        decode_shifted.append(tmp_entry)
+    return numpy.array(decode_shifted)
+
+
+def _double_decoding(dec1, dec2):
```

Comment:
Again, let's talk about whether this should be a "private" function. Also, I would prefer if you had slightly more complete names like "decoder_1, decoder_2" for the arguments of functions.